### PR TITLE
DOC/TYP: Fixed typing and doc

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] xref https://github.com/pandas-dev/pandas/pull/33263#pullrequestreview-389118991
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/aggregation.py [hunk 1]
```diff
@@ -40,7 +43,9 @@ def is_multi_agg_with_relabel(**kwargs) -> bool:
     )
 
 
-def normalize_keyword_aggregation(kwargs: dict) -> Tuple[dict, List[str], List[int]]:
+def normalize_keyword_aggregation(
+    kwargs: dict,
+) -> Tuple[DefaultDict, Tuple[Hashable, ...], "np.ndarray"]:
```
Comment (WillAyd) [inline pandas/core/aggregation.py:48]: What issue is this fixing?

Comment (simonjayhawkins) [reply pandas/core/aggregation.py:48]: see https://github.com/pandas-dev/pandas/pull/33263#discussion_r403133623

Comment (simonjayhawkins) [reply pandas/core/aggregation.py:48]: @MomIsBestFriend can you git bisect or blame to find out what's the story behind the here.

Comment (MomIsBestFriend) [reply pandas/core/aggregation.py:48]: @simonjayhawkins It looks like no one can agree on the placement of this function, as it's moved around constantly.

---

But it looks like parts of the changes are made in #30497 while migrating from ```orderedict``` to ```defaultdict```

File: pandas/core/aggregation.py [hunk 2]
```diff
@@ -40,7 +43,9 @@ def is_multi_agg_with_relabel(**kwargs) -> bool:
     )
 
 
-def normalize_keyword_aggregation(kwargs: dict) -> Tuple[dict, List[str], List[int]]:
+def normalize_keyword_aggregation(
+    kwargs: dict,
```
Comment (simonjayhawkins) [inline pandas/core/aggregation.py:47]: this should be Dict from typing. can you see if you can also add type parameters to Dict.

File: pandas/core/aggregation.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/aggregation.py:48]: normalize_keyword_aggregation is only called by aggregate in core\groupby\generic.py. 

the core\groupby\generic.py module is currently excluded by check_untyped_defs and aggregate does not have type hints.

maybe worth typing aggregate as well to ensure consistency.

Comment (MomIsBestFriend) [reply pandas/core/aggregation.py]: @simonjayhawkins Nice! thanks to that tip I could annotate with much more specific type hints.

File: pandas/core/groupby/generic.py [hunk 3]
```diff
@@ -909,7 +911,9 @@ class DataFrameGroupBy(GroupBy[DataFrame]):
         axis="",
     )
     @Appender(_shared_docs["aggregate"])
-    def aggregate(self, func=None, *args, **kwargs):
+    def aggregate(
+        self, func: Optional[DefaultDict[str, List[Scalar]]] = None, *args, **kwargs
```
Comment (jreback) [inline pandas/core/groupby/generic.py:915]: this can also be a list / dict of callables no?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
@MomIsBestFriend can you rebase (and address any actionable comments)

**Comment by simonjayhawkins** [comment]
Thanks @MomIsBestFriend closing as stale. ping to reopen.

**Pull request closed without merging**